zipline/finance/blotter/core.py

```diff
@@ -0,0 +1,111 @@
+from functools import partial
+from .blotter import Blotter
+from zipline.utils.compat import mappingproxy
+
+
+class BlotterClassDispatcher(object):
+    """
+    A class for registering and dispatching custom blotter classes.
+
+    Method of a global instance of this class are provided by
+    zipline.utils.calendar_utils.
+
+    Parameters
+    ----------
+    blotter_factories : dict[str -> function]
+        Factories for lazy blotter creation.
+    """
+    def __init__(self, blotter_factories):
+        self._blotter_factories = blotter_factories
+        self.blotter_factories = mappingproxy(self._blotter_factories)
+
+    def load(self, name):
+        """
+        Retrieves a blotter whose name is given.
+
+        Parameters
+        ----------
+        name : str
+            The name of the blotter class to be retrieved.
+
+        Returns
+        -------
+        blotter_class : zipline.finance.blotter.Blotter
+            The desired blotter class.
+        """
+        try:
+            return self._blotter_factories[name]
+        except KeyError:
+            raise ValueError("no blotter class registered as %r, options "
+                             "are: %r" % (name, sorted(self._blotter_factories
+                                                       )))
```

Comment:
Normally when we need to break a line to fit into the line limit, we would indent after the function call and list every argument on a new line, for example:

```python
            raise ValueError(
                "no blotter class registered as %r, options are: %r" % (
                    name,
                    sorted(self._blotter_factories),
                ),
            )
```

Most people have ample vertical space, and this make the argument and expression boundaries very clear.